sdk/state/payment.go

```diff
@@ -28,7 +28,7 @@ type CloseAgreementDetails struct {
 func (d CloseAgreementDetails) Equal(d2 CloseAgreementDetails) bool {
 	// TODO: Replace cmp.Equal with a hand written equals.
 	type CAD CloseAgreementDetails
-	return cmp.Equal(CAD(d), CAD(d2), cmp.AllowUnexported(keypair.FromAddress{}))
```

Comment:
Q: why are you removing `cmp.AllowUnexported(keypair.FromAddress{})` here?

Reply:
It's related to the use of the Equal function which was added to the keypair package in the go repo. When the Equal function wasn't present the cmp package uses reflection to check each exported field. The keypairs have an unexported field and this option tells the cmp package to compare the unexported fields of that type. Now that the Equal function exists the cmp package will use the Equal function instead of reflection.

Reply:
Sounds good, thanks for explaining.